File: test model zero bias.jl
using Flux

# Input size and hidden layer size
n_inputs = 4
n_hidden = 64

# Train the model with bias
train_model = Chain(
    Dense(n_inputs, n_hidden, relu),
    Dense(n_hidden, 1)
)

# ... training process ...

# Create a copy of the trained model for evaluation
eval_model = deepcopy(train_model)

# Set the biases of all dense layers in the evaluation model to zero
for layer in eval_model.layers
    if layer isa Dense
        Flux.params(layer)[2] .= zeros(size(Flux.params(layer)[2]))
    end
end

# Evaluate the model
prediction = eval_model([1,1,1,1])

File: test model zero bias.py
using Flux

# Input size
n_inputs = 4

# Train the model with bias
train_model = Chain(
    Dense(n_inputs, 1)
)

# ... training process ...

# Create a new model for evaluation with bias set to zero
eval_model = Chain(
    Dense(n_inputs, 1, bias=zeros(1))
)

# Copy the weights from the trained model
Flux.params(eval_model)[1] .= Flux.params(train_model)[1]

# Evaluate the model
prediction = eval_model(x)
